code/optimize/src/main/java/com/adobe/marketing/mobile/optimize/Offer.java

```diff
@@ -304,16 +317,35 @@ public void tapped() {
      * along with any additional XDM, free-form data, and override dataset identifier.
      *
      * @return {@code Map<String, Object>} containing the XDM data for the proposition interaction.
-     * @see OptimizeUtils#generateInteractionXdm(String, List, Boolean)
+     * @see OptimizeUtils#generateInteractionXdm(String, List)
      */
     public Map<String, Object> generateDisplayInteractionXdm() {
         if (propositionReference == null || propositionReference.get() == null) {
             return null;
         }
         return OptimizeUtils.generateInteractionXdm(
                 OptimizeConstants.JsonValues.EE_EVENT_TYPE_PROPOSITION_DISPLAY,
-                Collections.singletonList(propositionReference.get()),
-                true);
+                Collections.singletonList(propositionReference.get()));
+    }
+
+    /**
+     * Generates a map containing XDM formatted data for {@code Experience Event -
+     * OptimizeProposition Interactions} field group from this {@code OptimizeProposition} item.
+     *
+     * <p>The returned XDM data does contain the {@code eventType} for the Experience Event with
+     * value {@code decisioning.propositionDisplay}.
+     *
+     * <p>Note: The Edge sendEvent API can be used to dispatch this data in an Experience Event
+     * along with any additional XDM, free-form data, and override dataset identifier.
+     *
+     * @param offers {@code List<Offer>} containing offer data.
+     * @return {@code Map<String, Object>} containing the XDM data for the proposition interaction.
+     * @see OptimizeUtils#generateInteractionXdm(String, List)
+     */
+    public static Map<String, Object> generateDisplayInteractionXdm(List<Offer> offers) {
+        return OptimizeUtils.generateInteractionXdm(
+                OptimizeConstants.JsonValues.EE_EVENT_TYPE_PROPOSITION_DISPLAY,
+                Objects.requireNonNull(toUniquePropositionsList(offers)));
```

Comment:
`Objects.requireNonNull` would throw a null pointer exception if `toUniquePropositionsList` returns null, which would cause a RuntimeException and app will crash. Can we handle this gracefully? Either we can throw the exception so the app can handle it or we can do a null check for result of `toUniquePropositionsList` and fail fast without calling `OptimizeUtils.generateInteractionXdm`

Reply:
updated @spoorthipujariadobe